src/vcpkg/commands.update-baseline.cpp

```diff
@@ -0,0 +1,146 @@
+#include <vcpkg/base/files.h>
+#include <vcpkg/base/messages.h>
+
+#include <vcpkg/commands.update-baseline.h>
+#include <vcpkg/configuration.h>
+#include <vcpkg/vcpkgcmdarguments.h>
+#include <vcpkg/vcpkgpaths.h>
+
+namespace
+{
+    namespace msg = vcpkg::msg;
+    DECLARE_AND_REGISTER_MESSAGE(UpdateBaselineNoConfiguration, (), "", "There is no configuration file to update.");
+    DECLARE_AND_REGISTER_MESSAGE(UpdateBaselineGitError,
+                                 (msg::url),
+                                 "",
+                                 "Git failed to fetch remote repository {url}:");
+}
+
+namespace vcpkg::Commands
+{
+    static const CommandStructure COMMAND_STRUCTURE{
+        create_example_string("x-update-baseline"),
+        0,
+        0,
+    };
+
+    static constexpr StringLiteral builtin_git_url = "https://github.com/microsoft/vcpkg";
+
+    static Optional<std::string> get_baseline_from_git_repo(const VcpkgPaths& paths, StringView url)
+    {
+        auto res = paths.git_fetch_from_remote_registry(url, "HEAD");
+        if (res.has_value())
+        {
+            return std::move(*res.get());
+        }
+        else
+        {
+            msg::println(Color::warning, msgUpdateBaselineGitError, msg::url = url);
+            msg::write_unlocalized_text_to_stdout(Color::warning, res.error());
+            return nullopt;
+        }
+    }
+
+    static Optional<std::string> get_latest_baseline(const VcpkgPaths& paths, const RegistryConfig& r)
```

Comment:
I think this code should live inside `configuration.cpp` to minimize the number of files that must be aware of different flavors of registries. It should offer something like

```cpp
/// <returns>nullopt on success</returns>
Optional<LocalizedString> update_baselines(const VcpkgPaths& paths, RegistryConfig& config_to_update);
```

To update builtin_baseline, this will have to synthesize a temporary config object, but I think that's pretty simple and clean.